src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,465 @@
+        const { availableOrganisms, ...rest } = INITIAL_STATE;
+        this.setState({
+            ...rest,
+            showModal: true,
+        });
+    };
+
+    closeModal = () => {
+        this.setState({ showModal: false });
+    };
+
+    handleOpenModal = () => {
+        console.log(this.state);
+        this.openModal();
+    };
+
+    handleCloseModal = () => {
+        this.closeModal();
+    };
+
+    getNumFeatures = () => {
+        return this.featureMetadata.genes.length;
+    };
+
+    getTopNumFeaturesArray = () => {
+        return [
+            this.getNumFeatures() < 100 ? this.getNumFeatures() : 100,
+            200,
+            300,
+            400,
+            500,
+        ].filter((topNumFeaturesValue) =>
+            topNumFeaturesValue <= this.getNumFeatures() ? true : false
+        );
+    };
+
+    handleSelectOrganism = (e, { value }) => {
+        this.setState({ selectedOrganism: value, gProfilerURL: null });
+    };
+
+    handleSelectSortBy = (e, { value }) => {
+        this.setState({ selectedSortBy: value, gProfilerURL: null });
+    };
+
+    handleChangeToken = (e, { value }) => {
+        this.setState({
+            selectedOrganism: null,
+            gProfilerToken: value,
+            gProfilerURL: null,
+        });
+    };
+
+    handleClickGotoGProfilerURL = () => {
+        window.open(this.state.gProfilerURL);
+    };
+
+    buildMetricTable = () => {
+        return this.featureMetadata.genes.map((gene: string, idx: number) => {
+            return {
+                gene: gene,
+                ...this.featureMetadata.metrics.reduce(
+                    (accumalatedMetrics, metric) => ({
+                        ...accumalatedMetrics,
+                        [metric.accessor]: metric.values[idx],
+                    }),
+                    {}
+                ),
+            };
+        });
+    };
+
+    buildFeatureQuery = (topNumFeatures, sortedFeatureMetricTable) => {
+        return topNumFeatures
+            .map((topNumFeaturesElement) => {
+                return {
+                    numFeatures: topNumFeaturesElement,
+                    topSortedFeatures: sortedFeatureMetricTable
+                        .slice(0, topNumFeaturesElement)
+                        .reduce(
+                            (acc, sortedFeatureMetricTableRow) => [
+                                ...acc,
+                                sortedFeatureMetricTableRow['gene'],
+                            ],
+                            []
+                        )
+                        .join('\n'),
+                };
+            })
+            .reduce((acc, el) => {
+                return (
+                    acc +
+                    `>Top_${el.numFeatures}` +
+                    '\n' +
+                    el.topSortedFeatures +
+                    '\n'
+                );
+            }, '');
+    };
+
+    buildGProfilerLink = (organism: string, query: string) => {
```

Comment:
Rather than have two buttons, wouldn't it make sense for there to be one button, which first builds the link and then opens the URL? I can't think of a reason when a user would want to make the URL without visiting.

All of the logic would remain the same, the same checks etc can be performed, but for the user it is nicer to only press one button.

Reply:
Yeah makes sense, I'll make it one